### DISCUSSION TITLE
Custom reporter support

### DISCUSSION THREAD

**Discussion by mongolyy** [biomejs/biome#4775] [Ideas]
## Overview

I would like Biome to support user-defined custom reporters.

## Background / Motivation

- I use [reviewdog](https://github.com/reviewdog/reviewdog). Reviewdog requires diagnostic messages to be formatted in its own [RDFormat](https://github.com/reviewdog/reviewdog?tab=readme-ov-file#reviewdog-diagnostic-format-rdformat). Currently, Biome does not provide a way to extend or override the output format easily.
- ESLint offers the ability to specify a custom format via its [-f / --format option](https://eslint.org/docs/latest/use/command-line-interface#-f---format). Users who have leveraged custom formatters in ESLint would also benefit from a similar feature in Biome.

**Comment by ematipico** [comment]
Biome already has different different reporters: https://biomejs.dev/reference/reporters/

> --reporter=<json|json-pretty|github|junit|summary|gitlab>

**Reply by mongolyy** [reply]
Thank you for your response.

I know there are multiple reporters available, but I’d like to define my own output format for the reporter.

Specifically, I want to output a custom format called “[RDFormat](https://github.com/reviewdog/reviewdog/tree/master/proto/rdf#rdjson)” for the tool I’m using.
Something like this, for example: `--reporter=my-customize.js`

**Reply by dyc3** [reply]
As a workaround, you could use the json reporter and write a script to convert that to RDFormat. If there's something missing from the json reporter that you need, we would definitely want to address that.

**Reply by ematipico** [reply]
Yeah, we could add a new reporter :) for `rdjson`

**Reply by weyert** [reply]
I wish you could run multiple reporters at the same time. I would like to have `github` and `junit`. No I have to run it them separately. Currently, I am using a script like below to achieve it.  Also it appears you can't write the output to disk and need to manage it yourself. 

```
# Check if running in GitHub Actions
if [ "$GITHUB_ACTIONS" = "true" ]; then
    # Generate junit report only in CI
   mkdir -p coverage
    biome lint --reporter=junit > coverage/junit.xml
    
    # Output Github Actions format
    biome lint --reporter=github
else
    # Regular lint command for local development
    biome lint
fi
```

**Reply by arendjr** [reply]
@weyert why would you use a script at all for local development if all it does is call `biome lint`? You could also add it to `package.json` if you like, so people only need to do `yarn/pnpm lint` or `npm run lint`.

Then the only thing that remains for CI are the commands:

```sh
mkdir -p coverage
biome lint --reporter=junit > coverage/junit.xml
biome lint --reporter=github
```

To be frank that seems like a perfectly reasonable CI configuration to me.
